modules/passkey/contracts/4337/SafeECDSASignerLaunchpad.sol

```diff
@@ -12,168 +12,315 @@ import {ERC1271} from "../libraries/ERC1271.sol";
+        if (msg.sender != SUPPORTED_ENTRYPOINT) {
+            revert UnsupportedEntryPoint();
+        }
         _;
     }
 
-    receive() external payable {}
-
     /**
-     * @dev Performs pre-validation setup by storing the hash of the Safe setup (owners, threshold, etc)
-     * and executes an optional delegatecall to the `to` address.
-     * Requirements:
-     * - The function can only be called by the proxy contract.
-     * - The delegatecall to the `to` address must succeed.
-     * @param initHash The initialization hash.
-     * @param to The address to delegatecall.
-     * @param preInit The pre-initialization call data.
+     * @notice Accept transfers.
+     * @dev The launchpad accepts transfers to allow funding of the account in case it was deployed and initialized
+     * without pre-funding.
      */
-    function preValidationSetup(bytes32 initHash, address to, bytes calldata preInit) external onlyProxy {
-        _setInitHash(initHash);
-        if (to != address(0)) {
-            (bool success, ) = to.delegatecall(preInit);
-            require(success, "Pre-initialization failed");
-        }
-    }
+    receive() external payable {}
 
     /**
-     * @notice This function generates a unique initialization hash for a Safe.
-     * @dev The hash is generated using the keccak256 hash function and the EIP-712 standard.
+     * @notice Sets up the account with the specified initialization parameters.
+     * @dev This function can only be called by a proxy contract that has not yet been initialized. Internally, it calls
+     * the Safe singleton setup function on the account with some pre-determined parameters. In particular, it uses a
+     * fixed ownership structure for the deployed Safe.
      * @param singleton The singleton to evolve into during the setup.
-     * @param signerFactory The unique signer factory to use for creating an owner.
+     * @param signerFactory The custom ECDSA signer factory to use for creating an owner.
      * @param signerX The X coordinate of the signer's public key.
      * @param signerY The Y coordinate of the signer's public key.
      * @param signerVerifier The address of the contract that verifies the signer's signature.
-     * @param setupTo The contract to delegatecall during setup.
-     * @param setupData The calldata for the setup delegatecall.
-     * @param fallbackHandler The fallback handler to initialize the Safe with.
-     * @return initHash The unique initialization hash for the Safe.
+     * @param initializer The Safe setup initializer address.
+     * @param initializerData The calldata for the setup `DELEGATECALL`.
+     * @param fallbackHandler Handler for fallback calls to the Safe.
      */
-    function getInitHash(
+    function setup(
         address singleton,
         address signerFactory,
         uint256 signerX,
         uint256 signerY,
         address signerVerifier,
```

Comment:
idea for multiple owners: make this an array of structs and add a separate ecdsa owner array (just realised that the main bottleneck is still the signature verification but still leacing this here)

In general I like the direction of where this is going.

Reply:
Captured in issue #399